File: backend-server/application/src/main/java/com/apitable/workspace/entity/DatasheetEntity.java


package com.apitable.workspace.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * Workbench - Data Table.
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(keepGlobalPrefix = true, value = "datasheet")
public class DatasheetEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * Primary key.
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * Custom ID.
     */
    private String dstId;

    /**
     * Datasheet Node Id(link#xxxx_node#node_id).
     */
    private String nodeId;

    /**
     * Name.
     */
    private String dstName;

    /**
     * Space ID(link#xxxx_space#space_id).
     */
    private String spaceId;

    /**
     * Creator.
     */
    private Long creator;

    /**
     * Version No.
     */
    private Long revision;

    /**
     * Delete tag(0:No,1:Yes).
     */
    @TableLogic
    private Boolean isDeleted;

    /**
     * Create User.
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * Last Update User.
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * Create Time.
     */
    private LocalDateTime createdAt;

    /**
     * Update Time.
     */
    private LocalDateTime updatedAt;

}

File: packages/room-server/src/database/datasheet/entities/datasheet.entity.ts


import { BaseEntity } from 'shared/entities/base.entity';
import { Column, Entity } from 'typeorm';

/**
 * Workbench-Datasheet
 */
@Entity('datasheet')
export class DatasheetEntity extends BaseEntity {
  @Column({
    name: 'dst_id',
    nullable: true,
    unique: true,
    comment: 'datasheet ID',
    length: 50,
  })
  dstId?: string;

  @Column({
    name: 'node_id',
    nullable: true,
    comment: 'node ID (association#node#node_id)',
    length: 50,
  })
  nodeId?: string;

  @Column({
    name: 'dst_name',
    nullable: true,
    comment: 'datasheet name',
    length: 255,
  })
  dstName!: string;

  @Column({
    name: 'space_id',
    nullable: true,
    comment: 'space ID(related#space#space_id)',
    length: 50,
  })
  spaceId?: string;

  @Column({
    name: 'creator',
    nullable: true,
    comment: 'creator ID',
  })
  creator?: string;

  @Column({
    name: 'revision',
    nullable: true,
    comment: 'revision',
    unsigned: true,
    default: () => 0,
    type: 'bigint',
    width: 20,
  })
  revision!: number;
}
